[qt5-base] build failure

### Operating system

kali Linux

### Compiler

GCC 13

### Steps to reproduce the behavior

```Shell
` ./vcpkg install cgal[qt] --binarysource=clear --clean-after-build`
```


### Failure logs

Computing installation plan...
The following packages will be built and installed:
    cgal[core,qt]:x64-linux@5.6
  * libwebp[core,libwebpmux,nearlossless,simd]:x64-linux@1.3.2
  * qt5-3d:x64-linux@5.15.12
  * qt5-base:x64-linux@5.15.12
  * qt5-declarative[core,platform-default-features]:x64-linux@5.15.12
  * qt5-gamepad:x64-linux@5.15.12
  * qt5-imageformats[core,tiff,webp]:x64-linux@5.15.12
  * qt5-script:x64-linux@5.15.12
  * qt5-svg:x64-linux@5.15.12
  * qt5-tools:x64-linux@5.15.12
  * qt5-xmlpatterns:x64-linux@5.15.12
  * tiff[core,jpeg,lzma,zip]:x64-linux@4.6.0#1
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-linux...
Installing 1/12 qt5-base:x64-linux@5.15.12...
Building qt5-base:x64-linux@5.15.12...
CMake Warning at ports/qt5-base/portfile.cmake:4 (message):
  qt5-base currently requires some packages from the system package manager.
  They can be installed on Ubuntu systems via sudo apt-get install
  '^libxcb.*-dev' libx11-xcb-dev libgl1-mesa-dev libxrender-dev libxi-dev
  libxkbcommon-dev libxkbcommon-x11-dev.  For more information, see
  https://doc.qt.io/qt-5/linux.html and
  https://doc.qt.io/qt-5/linux-requirements.html
Call Stack (most recent call first):
  scripts/ports.cmake:170 (include)


-- Figuring out qt target mkspec. Target arch x64
-- Target mkspec set to: linux-g++
-- Host mkspec not set. Qt's own buildsystem will try to figure out the host system
-- Using cached qtbase-everywhere-opensource-src-5.15.12.tar.xz.
-- Extracting source /home/bazunzhi/Environments/vcpkg/downloads/qtbase-everywhere-opensource-src-5.15.12.tar.xz
-- Applying patch patches/CVE-2023-24607-qtbase-5.15.diff
-- Applying patch patches/CVE-2023-32762-qtbase-5.15.diff
-- Applying patch patches/CVE-2023-32763-qtbase-5.15.diff
-- Applying patch patches/CVE-2023-33285-qtbase-5.15.diff
-- Applying patch patches/CVE-2023-34410-qtbase-5.15.diff
-- Applying patch patches/CVE-2023-37369-qtbase-5.15.diff
-- Applying patch patches/CVE-2023-38197-qtbase-5.15.diff
-- Applying patch patches/CVE-2023-43114-5.15.patch
-- Applying patch patches/0001-CVE-2023-51714-qtbase-5.15.diff
-- Applying patch patches/0002-CVE-2023-51714-qtbase-5.15.diff
-- Applying patch patches/winmain_pro.patch
-- Applying patch patches/windows_prf.patch
-- Applying patch patches/qt_app.patch
-- Applying patch patches/gui_configure.patch
-- Applying patch patches/xlib.patch
-- Applying patch patches/egl.patch
-- Applying patch patches/qtbug_96392.patch
-- Applying patch patches/mysql_plugin_include.patch
-- Applying patch patches/mysql-configure.patch
-- Applying patch patches/cocoa.patch
-- Applying patch patches/xcode-15.patch
-- Applying patch patches/patch-qtbase-memory_resource.diff
-- Applying patch patches/Qt5BasicConfig.patch
-- Applying patch patches/Qt5PluginTarget.patch
-- Applying patch patches/create_cmake.patch
-- Applying patch patches/Qt5GuiConfigExtras.patch
-- Applying patch patches/fix_angle.patch
-- Applying patch patches/mingw9.patch
-- Using source at /home/bazunzhi/Environments/vcpkg/buildtrees/qt5-base/src/5.15.12-1182f234b9.clean
-- Getting CMake variables for x64-linux
-- Configuring x64-linux-dbg
-- Configuring x64-linux-dbg done
-- Configuring x64-linux-rel
-- Configuring x64-linux-rel done
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/bin/make -j13
    Working Directory: /home/bazunzhi/Environments/vcpkg/buildtrees/qt5-base/x64-linux-dbg
    See logs for more information:
      /home/bazunzhi/Environments/vcpkg/buildtrees/qt5-base/build-x64-linux-dbg-out.log
      /home/bazunzhi/Environments/vcpkg/buildtrees/qt5-base/build-x64-linux-dbg-err.log

Call Stack (most recent call first):
  ports/qt5-base/cmake/install_qt.cmake:86 (vcpkg_execute_build_process)
  ports/qt5-base/portfile.cmake:401 (install_qt)
  scripts/ports.cmake:170 (include)


error: building qt5-base:x64-linux failed with: BUILD_FAILED



### Additional context

[build-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/13799900/build-x64-linux-dbg-out.log)
[build-x64-linux-dbg-err.log](https://github.com/microsoft/vcpkg/files/13799903/build-x64-linux-dbg-err.log)

--- a/ports/qt5-base/patches/XKB_KEY_dead_lowline.patch
+++ b/ports/qt5-base/patches/XKB_KEY_dead_lowline.patch
@@ -1,0 +1,18 @@
+diff --git a/src/platformsupport/input/xkbcommon/qxkbcommon.cpp b/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
+index 7bfacf4..ded358a 100644
+--- a/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
++++ b/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
+@@ -273,11 +273,13 @@ static constexpr const auto KeyTbl = qMakeArray(
+         Xkb2Qt<XKB_KEY_dead_small_schwa,        Qt::Key_Dead_Small_Schwa>,
+         Xkb2Qt<XKB_KEY_dead_capital_schwa,      Qt::Key_Dead_Capital_Schwa>,
+         Xkb2Qt<XKB_KEY_dead_greek,              Qt::Key_Dead_Greek>,
++#ifdef XKB_KEY_dead_lowline
+         Xkb2Qt<XKB_KEY_dead_lowline,            Qt::Key_Dead_Lowline>,
+         Xkb2Qt<XKB_KEY_dead_aboveverticalline,  Qt::Key_Dead_Aboveverticalline>,
+         Xkb2Qt<XKB_KEY_dead_belowverticalline,  Qt::Key_Dead_Belowverticalline>,
+         Xkb2Qt<XKB_KEY_dead_longsolidusoverlay, Qt::Key_Dead_Longsolidusoverlay>,
+ 
++#endif
+         // Special keys from X.org - This include multimedia keys,
+         // wireless/bluetooth/uwb keys, special launcher keys, etc.
+         Xkb2Qt<XKB_KEY_XF86Back,                Qt::Key_Back>,

--- a/ports/qt5-base/portfile.cmake
+++ b/ports/qt5-base/portfile.cmake
@@ -80,6 +80,7 @@ qt_download_submodule(  OUT_SOURCE_PATH SOURCE_PATH
                             patches/Qt5GuiConfigExtras.patch   # Patches the library search behavior for EGL since angle is not build with Qt
                             patches/fix_angle.patch            # Failed to create OpenGL context for format QSurfaceFormat ...
                             patches/mingw9.patch               # Fix compile with MinGW-W64 9.0.0: Redefinition of 'struct _FILE_ID_INFO'
+                            patches/XKB_KEY_dead_lowline.patch # https://bugreports.qt.io/browse/QTBUG-117950
                     )
 
 # Remove vendored dependencies to ensure they are not picked up by the build

--- a/ports/qt5-base/vcpkg.json
+++ b/ports/qt5-base/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "qt5-base",
   "version": "5.15.12",
+  "port-version": 1,
   "description": "Qt5 Application Framework Base Module. Includes Core, GUI, Widgets, Networking, SQL, Concurrent and other essential qt components.",
   "homepage": "https://www.qt.io/",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6946,7 +6946,7 @@
     },
     "qt5-base": {
       "baseline": "5.15.12",
-      "port-version": 0
+      "port-version": 1
     },
     "qt5-canvas3d": {
       "baseline": "0",

--- a/versions/q-/qt5-base.json
+++ b/versions/q-/qt5-base.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "2932bd39b46da13edde2cf5715b0308c53b1d426",
+      "version": "5.15.12",
+      "port-version": 1
+    },
+    {
       "git-tree": "e2a320e35b03c115bc674b6006f7a8a9907835b0",
       "version": "5.15.12",
       "port-version": 0

